### PR TITLE
XJC: External parsing is disabled. Cannot parse URI

### PR DESCRIPTION
**Pull request by jbescos**
I tried with:
jdk1.8.0_231
jdk-11.0.7

Executing the Websockets I was getting all the time this error with the arguments -Djavax.xml.accessExternalStylesheet=all -Djavax.xml.accessExternalDTD=all -Djavax.xml.accessExternalSchema=all -Djakarta.xml.accessExternalStylesheet=all -Djakarta.xml.accessExternalDTD=all -Djakarta.xml.accessExternalSchema=all -DenableExternalEntityProcessing=true:

```
bind.dtds:
dropping /home/jbescos/workspace/jakartaee-tck/lib/websockettck.jar from path as it doesn't exist
dropping /home/jbescos/programs/glassfish6/glassfish/modules/tyrus-websocket-core.jar from path as it doesn't exist
dropping /home/jbescos/programs/glassfish6/glassfish/modules/tyrus-container-grizzly.jar from path as it doesn't exist
      [xjc] build id of XJC is 3.0.0-M3
      [xjc] Checking timestamp of /home/jbescos/workspace/jakartaee-tck/lib/dtds/sun-application_6_0-0.dtd
      [xjc] Checking timestamp of /home/jbescos/workspace/jakartaee-tck/src/com/sun/ts/lib/implementation/sun/javaee/runtime/jaxb-cust.xml
      [xjc] Checking timestamp of /home/jbescos/workspace/jakartaee-tck/lib/dtds/sun-application_6_0-0.dtd
      [xjc] the last modified time of the inputs is  1579786959000
      [xjc] the last modified time of the outputs is -9223372036854775808
      [xjc] parsing a schema...
      [xjc] [ERROR] External parsing is disabled. Cannot parse URI: file:/home/jbescos/workspace/jakartaee-tck/lib/dtds/sun-application_6_0-0.dtd
      [xjc] unknown location
      [xjc] 
      [xjc] Failed to parse a schema.
      [xjc] Command invoked: xjc /home/jbescos/programs/java/jdk1.8.0_231/jre/bin/java -classpath /home/jbescos/workspace/jakartaee-tck/lib/jaxb-xjc.jar:/home/jbescos/workspace/jakartaee-tck/lib/jaxb-core.jar:/home/jbescos/workspace/jakartaee-tck/lib/jaxb-impl.jar:/home/jbescos/workspace/jakartaee-tck/lib/tsharness.jar:/home/jbescos/workspace/jakartaee-tck/lib/javatest.jar:/home/jbescos/programs/apache-ant-1.9.14/lib/ant.jar:/home/jbescos/workspace/jakartaee-tck/classes:/home/jbescos/programs/glassfish6/glassfish/modules/tyrus-client.jar:/home/jbescos/programs/glassfish6/glassfish/modules/tyrus-core.jar:/home/jbescos/programs/glassfish6/glassfish/modules/glassfish-grizzly-extra-all.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2-core.jar:/home/jbescos/programs/glassfish6/glassfish/modules/glassfish-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2-utils.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jakarta.annotation-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jakarta.inject-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/aopalliance-repackaged.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2-locator.jar:/home/jbescos/programs/glassfish6/glassfish/modules/javassist.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2-runlevel.jar:/home/jbescos/programs/glassfish6/glassfish/modules/class-model.jar:/home/jbescos/programs/glassfish6/glassfish/modules/asm.jar:/home/jbescos/programs/glassfish6/glassfish/modules/asm-analysis.jar:/home/jbescos/programs/glassfish6/glassfish/modules/asm-commons.jar:/home/jbescos/programs/glassfish6/glassfish/modules/asm-tree.jar:/home/jbescos/programs/glassfish6/glassfish/modules/asm-util.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jakarta.xml.bind-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jaxb-osgi.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jakarta.activation.jar:/home/jbescos/programs/glassfish6/glassfish/modules/scattered-archive-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hk2-config.jar:/home/jbescos/programs/glassfish6/glassfish/modules/tiger-types.jar:/home/jbescos/programs/glassfish6/glassfish/modules/hibernate-validator.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jakarta.validation-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/jboss-logging.jar:/home/jbescos/programs/glassfish6/glassfish/modules/classmate.jar:/home/jbescos/programs/glassfish6/glassfish/modules/nucleus-grizzly-all.jar:/home/jbescos/programs/glassfish6/glassfish/modules/config-types.jar:/home/jbescos/programs/glassfish6/glassfish/modules/management-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/internal-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/config-api.jar:/home/jbescos/programs/glassfish6/glassfish/modules/common-util.jar:/home/jbescos/programs/glassfish6/glassfish/modules/tyrus-server.jar:/home/jbescos/programs/glassfish6/glassfish/modules/tyrus-container-servlet.jar:/home/jbescos/programs/glassfish6/glassfish/modules/tyrus-container-grizzly-client.jar:/home/jbescos/programs/glassfish6/glassfish/modules/tyrus-spi.jar:/home/jbescos/programs/glassfish6/glassfish/modules/glassfish-corba-omgapi.jar:/home/jbescos/workspace/jakartaee-tck/lib/sigtest.jar:/home/jbescos/workspace/jakartaee-tck/lib/commons-httpclient-3.1.jar:/home/jbescos/workspace/jakartaee-tck/lib/commons-logging-1.1.3.jar:/home/jbescos/workspace/jakartaee-tck/lib/commons-codec-1.9.jar:/home/jbescos/programs/java/jdk1.8.0_231/lib/tools.jar com.sun.tools.xjc.XJCFacade -d /home/jbescos/workspace/jakartaee-tck/src -p com.sun.ts.lib.implementation.sun.javaee.runtime.app -dtd /home/jbescos/workspace/jakartaee-tck/lib/dtds/sun-application_6_0-0.dtd -b file:/home/jbescos/workspace/jakartaee-tck/bin/xml/../../src/com/sun/ts/lib/implementation/sun/javaee/runtime/jaxb-cust.xml
      [xjc] failure in the XJC task. Use the Ant -verbose switch for more details
  [antcall] Exiting /home/jbescos/workspace/jakartaee-tck/install/websocket/bin/build.xml.
      [ant] Exiting /home/jbescos/workspace/jakartaee-tck/install/websocket/bin/build.xml.
BUILD FAILED
/home/jbescos/workspace/jakartaee-tck/bin/xml/ts.import.xml:64: The following error occurred while executing this line:
/home/jbescos/workspace/jakartaee-tck/bin/xml/ts.top.import.xml:391: The following error occurred while executing this line:
/home/jbescos/workspace/jakartaee-tck/bin/xml/ts.top.import.xml:690: xjc failed
	at com.sun.tools.xjc.XJCBase.execute(XJCBase.java:679)
	at com.sun.tools.xjc.XJC2Task.execute(XJC2Task.java:25)
	at com.sun.istack.tools.ProtectedTask.execute(ProtectedTask.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:352)
	at org.apache.tools.ant.Target.execute(Target.java:437)
	at org.apache.tools.ant.Target.performTasks(Target.java:458)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1406)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:36)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1261)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
	at org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:105)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:352)
	at org.apache.tools.ant.Target.execute(Target.java:437)
	at org.apache.tools.ant.Target.performTasks(Target.java:458)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1406)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:36)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1261)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:352)
	at org.apache.tools.ant.Target.execute(Target.java:437)
	at org.apache.tools.ant.Target.performTasks(Target.java:458)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1406)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1377)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1261)
	at org.apache.tools.ant.Main.runBuild(Main.java:857)
	at org.apache.tools.ant.Main.startAnt(Main.java:236)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:287)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:112)

```
The task needs the jvm argument -DenableExternalEntityProcessing=true

### CODE REVIEW

**Comment by jbescos** [comment]
@lukasj could you review it?

File: install/jsonb/bin/ts.jte [hunk 1]
```diff
@@ -63,11 +63,15 @@ jsonb.classes=
 # @ts.classpath         --  Classes used to build the JSON-B tests
 # @ts.run.classpath     --  Classes needed by test run
 ####################################################################
+glassfish.modules.path=/home/jbescos/programs/glassfish6/glassfish/modules
+missing.jars=${ts.home}/lib
+ts.required.lib=${pathsep}${glassfish.modules.path}/yasson.jar${pathsep}${missing.jars}/jakarta.inject-api-2.0.0-RC4.jar${pathsep}${glassfish.modules.path}/jakarta.json.jar${pathsep}${glassfish.modules.path}/jakarta.json.bind-api.jar${pathsep}${glassfish.modules.path}/jakarta.servlet-api.jar
```
Comment (lukasj) [inline install/jsonb/bin/ts.jte:68]: how is jakarta.inject-api-2.0.0-RC4.jar different from ${glassfish.modules.path}/jakarta.inject-api.jar ?

File: install/websocket/bin/ts.jte [hunk 2]
```diff
@@ -77,7 +77,7 @@ javatest.timeout.factor=1.0
 ## Settings for Vendor WebSocket Implementation
 #  @web.home The location of the vendor's WebSocket Implementation
 ########################################################################
-web.home=
+web.home=/home/jbescos/programs/glassfish5/glassfish
```
Comment (lukasj) [inline install/websocket/bin/ts.jte:80]: this is definitely not something you want to put in

File: install/websocket/bin/ts.jte [hunk 3]
```diff
@@ -101,6 +101,7 @@ command.testExecute=com.sun.ts.lib.harness.ExecTSTestCmd \
                     -Djavax.net.ssl.keyStorePassword=changeit \
                     -Djavax.net.ssl.keyStore=${bin.dir}/certificates/clientcert.jks \
                     -Djavax.net.ssl.trustStore=${web.home}/domains/domain1/config/cacerts.jks \
+                    -DenableExternalEntityProcessing=true -Djavax.xml.accessExternalStylesheet=all -Djavax.xml.accessExternalDTD=all -Djavax.xml.accessExternalSchema=all -Djakarta.xml.accessExternalStylesheet=all -Djakarta.xml.accessExternalDTD=all -Djakarta.xml.accessExternalSchema=all \
```
Comment (lukasj) [inline install/websocket/bin/ts.jte:104]: are you sure this is required?

File: install/websocket/bin/ts.jte [hunk 4]
```diff
@@ -174,8 +175,8 @@ porting.ts.url.class.1=com.sun.ts.lib.implementation.sun.common.SunRIURL
 # @webServerPort  port number of the Vendor's Java EE Web Server
 #
 ##########################################################################
-webServerHost=
-webServerPort=
+webServerHost=localhost
+webServerPort=8000
```
Comment (lukasj) [inline install/websocket/bin/ts.jte:180]: again, not something to be put in

File: install/websocket/bin/ts.jte [hunk 5]
```diff
@@ -205,11 +206,12 @@ ClientThreads=2
...
-
+websocket.classes=${web.home}/modules/tyrus-websocket-core.jar${pathsep}${web.home}/modules/tyrus-client.jar${pathsep}${web.home}/modules/tyrus-core.jar${pathsep}${web.home}/modules/tyrus-container-grizzly.jar${pathsep}${web.home}/modules/glassfish-grizzly-extra-all.jar${pathsep}${web.home}/modules/nucleus-grizzly-all.jar${pathsep}${web.home}/modules/tyrus-server.jar${pathsep}${web.home}/modules/tyrus-container-servlet.jar${pathsep}${web.home}/modules/tyrus-container-grizzly-client.jar${pathsep}${web.home}/modules/tyrus-spi.jar${pathsep}${web.home}/modules/glassfish-corba-omgapi.jar
+ts.custom.classpath=/home/jbescos/Downloads/jaxb-xjc-2.3.3.jar${pathsep}${web.home}/modules/jakarta.xml.bind-api.jar
```
Comment (lukasj) [inline install/websocket/bin/ts.jte:210]: so now I know where is your Downloads folder... :-D

**Comment by jbescos** [comment]
I mistakenly committed some files. This PR only applies for ts.top.import.xml. It should be fixed now.

**Comment by lukasj** [comment]
do not forget to fix sign-off

**Comment by scottmarlow** [comment]
@jbescos did you use the "git commit -s" or "git commit --amend --signoff" option?

**Comment by alwin-joseph** [comment]
The xjc jars were updated in the tck repo, and we had added this line : https://github.com/eclipse-ee4j/jakartaee-tck/blob/72e162ac3eaeeb4b115a717f40d19e803fc1ec0f/docker/build_jakartaeetck.sh#L81 to fix the same parsing issue for building platform TCK.

**Comment by jbescos** [comment]
> @jbescos did you use the "git commit -s" or "git commit --amend --signoff" option?

I did commit -s. Currently there is only one commit that contains the Signed-off-by. I don't see anything wrong in the commit, is there are issue that I don't realize?.

```
commit 6cac19c332106ae221a1ddbcd5d348fefa792c58 (HEAD -> xjc, origin/xjc)
Author: Jorge Bescos Gascon <jorge.bescos.gascon@oracle.com>
Date:   Wed Jun 24 17:11:14 2020 +0200

    XJC: External parsing is disabled. Cannot parse URI
    
    Signed-off-by: Jorge Bescos Gascon <jorge.bescos.gascon@oracle.com>
```

**Comment by scottmarlow** [comment]
@jbescos I see a message "the eclipsefdn/eca - The author(s) of the pull request is not covered by necessary legal...".

**Comment by jbescos** [comment]
@scottmarlow yes, you are right. I also have the same problem in another PR in jersey.

The reason is that I is in process of being a commiter in Jersey, and that triggered the next status:

> Thank you for providing your employer information, please allow 5 business days for processing before we send you the necessary paperwork.

**Comment by jbescos** [comment]
> The xjc jars were updated in the tck repo, and we had added this line :
> 
> https://github.com/eclipse-ee4j/jakartaee-tck/blob/72e162ac3eaeeb4b115a717f40d19e803fc1ec0f/docker/build_jakartaeetck.sh#L81
> to fix the same parsing issue for building platform TCK.

@alwin-joseph I think it is fine to add it also in the ant xml, so we can run it locally. Do you agree?.

**Comment by jbescos** [comment]
@scottmarlow I am legal developer again :)

**Comment by scottmarlow** [comment]
Thanks Jorge for the fix!

**Pull request merged**